### PR TITLE
fix: Fix the failing tests on fy-view-report-info component

### PR DESCRIPTION
**Pull request by arjunaj5**
### <samp>🤖 Generated by Copilot at 1de6ff5</samp>

This pull request updates the `fy-view-report-info` component and its test to use a service for getting the report summary data. It also adds mock data and test assertions for multiple currencies and reimbursable expenses.
https://app.clickup.com/t/86ctx56rq

### <samp>🤖 Generated by Copilot at 1de6ff5</samp>

> _We mock the data, we test the view_
> _We share the expenses, we face the truth_
> _Multiple currencies, reimbursable costs_
> _We are the coders, we rule the reports_

### <samp>🤖 Generated by Copilot at 1de6ff5</samp>

*  Add new mock data variable `expenseResponseData2` for testing report summary with multiple expenses ([link](https://github.com/fylein/fyle-mobile-app/pull/2559/files?diff=unified&w=0#diff-24d88c263ca5d224aa35b783df4640316640e6db32915015b6cce6ff83b7a05bR360-R361))
*  Import new mock data variable and shared expenses service in test file ([link](https://github.com/fylein/fyle-mobile-app/pull/2559/files?diff=unified&w=0#diff-5898bfb2270735be796155dc1dccda8aee20ea0b3e0194b1b9f2b769647a99bbL21-R28))
*  Create spy object for shared expenses service with summary methods ([link](https://github.com/fylein/fyle-mobile-app/pull/2559/files?diff=unified&w=0#diff-5898bfb2270735be796155dc1dccda8aee20ea0b3e0194b1b9f2b769647a99bbR45-R48))
*  Provide spy object as value for shared expenses service in test bed configuration ([link](https://github.com/fylein/fyle-mobile-app/pull/2559/files?diff=unified&w=0#diff-5898bfb2270735be796155dc1dccda8aee20ea0b3e0194b1b9f2b769647a99bbR62-R65))
*  Assign spy object to `sharedExpensesService` variable in test setup ([link](https://github.com/fylein/fyle-mobile-app/pull/2559/files?diff=unified&w=0#diff-5898bfb2270735be796155dc1dccda8aee20ea0b3e0194b1b9f2b769647a99bbR94))
*  Replace calls to transaction service methods with shared expenses service methods in component and test file ([link](https://github.com/fylein/fyle-mobile-app/pull/2559/files?diff=unified&w=0#diff-5898bfb2270735be796155dc1dccda8aee20ea0b3e0194b1b9f2b769647a99bbL124-R136), [link](https://github.com/fylein/fyle-mobile-app/pull/2559/files?diff=unified&w=0#diff-5898bfb2270735be796155dc1dccda8aee20ea0b3e0194b1b9f2b769647a99bbL162-R176), [link](https://github.com/fylein/fyle-mobile-app/pull/2559/files?diff=unified&w=0#diff-5898bfb2270735be796155dc1dccda8aee20ea0b3e0194b1b9f2b769647a99bbL173-R189))
*  Modify test description and expectation for `ionViewWillEnter` method to reflect multiple currencies and reimbursable amount ([link](https://github.com/fylein/fyle-mobile-app/pull/2559/files?diff=unified&w=0#diff-5898bfb2270735be796155dc1dccda8aee20ea0b3e0194b1b9f2b769647a99bbL138-R154), [link](https://github.com/fylein/fyle-mobile-app/pull/2559/files?diff=unified&w=0#diff-5898bfb2270735be796155dc1dccda8aee20ea0b3e0194b1b9f2b769647a99bbL152-R166))

## Clickup
Please add link here

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/shared/components/fy-view-report-info/fy-view-report-info.component.spec.ts [hunk 1]
```diff
@@ -135,12 +146,12 @@ describe('FyViewReportInfoComponent', () => {
       Reimbursable: 4600,
     });
     expect(component.getCCCAdvanceSummary).toHaveBeenCalledOnceWith(paymentModeSummaryMock, orgSettingsRes);
-    expect(transactionService.getCurrenyWiseSummary).toHaveBeenCalledOnceWith(expenseList);
+    expect(sharedExpensesService.getCurrenyWiseSummary).toHaveBeenCalledOnceWith(expenseResponseData);
     expect(component.amountCurrencyWiseDetails).toEqual(currencySummaryData);
-    expect(component.isForeignCurrency).toBe(false);
+    expect(component.isForeignCurrency).toBe(true);
```
Comment (arjunaj5) [inline src/app/shared/components/fy-view-report-info/fy-view-report-info.component.spec.ts:151]: mock data changed

File: src/app/shared/components/fy-view-report-info/fy-view-report-info.component.spec.ts [hunk 2]
```diff
@@ -149,33 +160,33 @@ describe('FyViewReportInfoComponent', () => {
...
     };
     const paymentModeSummaryMock = {
-      ccc: {
-        name: 'example',
-        key: 'key123',
-        amount: 4600,
+      reimbursable: {
+        name: 'Reimbursable',
+        key: 'reimbursable',
+        amount: 207000.78,
```
Comment (arjunaj5) [inline src/app/shared/components/fy-view-report-info/fy-view-report-info.component.spec.ts:166]: Mock data changed

File: src/app/shared/components/fy-view-report-info/fy-view-report-info.component.spec.ts [hunk 3]
```diff
@@ -149,33 +160,33 @@ describe('FyViewReportInfoComponent', () => {
...
       'Total Amount': 46040,
-      Reimbursable: 0,
+      Reimbursable: 207000.78,
     });
     expect(component.getCCCAdvanceSummary).toHaveBeenCalledOnceWith(paymentModeSummaryMock, orgSettingsRes);
-    expect(transactionService.getCurrenyWiseSummary).toHaveBeenCalledOnceWith(expenseList);
+    expect(sharedExpensesService.getCurrenyWiseSummary).toHaveBeenCalledOnceWith(expenseResponseData2);
     expect(component.amountCurrencyWiseDetails).toEqual(currencySummaryData);
-    expect(component.isForeignCurrency).toBe(false);
+    expect(component.isForeignCurrency).toBe(true);
```
Comment (arjunaj5) [inline src/app/shared/components/fy-view-report-info/fy-view-report-info.component.spec.ts:189]: mock data changed

**Review by jayfyle — APPROVED** [review]
Looks good to me

**Pull request merged**